File: app/src/main/java/cn/nekocode/baseframework/model/Weather.java
package cn.nekocode.baseframework.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by nekocode on 2015/7/17.
 */
public class Weather {
    @SerializedName("weatherinfo")
    private WeatherInfo weatherInfo;

    public static class WeatherInfo implements Parcelable, Cloneable {
        @SerializedName("city")
        private String city;

        public String getCity() {
            return city;
        }

        public void setCity(String city) {
            this.city = city;
        }

        @Override
        public int describeContents() {
            return 0;
        }

        @Override
        public void writeToParcel(Parcel dest, int flags) {
            dest.writeString(this.city);
        }

        public WeatherInfo() {
        }

        protected WeatherInfo(Parcel in) {
            this.city = in.readString();
        }

        public static final Creator<WeatherInfo> CREATOR = new Creator<WeatherInfo>() {
            public WeatherInfo createFromParcel(Parcel source) {
                return new WeatherInfo(source);
            }

            public WeatherInfo[] newArray(int size) {
                return new WeatherInfo[size];
            }
        };

        public WeatherInfo clone(){
            try {
                return (WeatherInfo) super.clone();
            } catch(CloneNotSupportedException ex){
                return null;
            }
        }
    }

    public WeatherInfo getWeatherInfo() {
        return weatherInfo;
    }

    public void setWeatherInfo(WeatherInfo weatherInfo) {
        this.weatherInfo = weatherInfo;
    }
}

File: app/src/main/java/cn/nekocode/baseframework/data/dto/Weather.kt
package cn.nekocode.baseframework.data.dto

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by nekocode on 2016/1/13.
 */
data class Weather(val city: String) : Parcelable {
    constructor(source: Parcel): this(source.readString())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeString(this.city)
    }

    companion object {
        @JvmField final val CREATOR: Parcelable.Creator<Weather> = object : Parcelable.Creator<Weather> {
            override fun createFromParcel(source: Parcel): Weather {
                return Weather(source)
            }

            override fun newArray(size: Int): Array<Weather?> {
                return arrayOfNulls(size)
            }
        }
    }
}